tests/test_sparkml_sql_transformer.py

```diff
@@ -0,0 +1,54 @@
+"""
+Tests Spark-ML SQLTransformer
+"""
+import unittest
+import warnings
+import json
+import pprint
+import numpy as np
+import torch
+from sklearn.datasets import load_iris
+from distutils.version import LooseVersion
+
+from hummingbird.ml._utils import sparkml_installed, pandas_installed
+from hummingbird.ml import convert
+
+if sparkml_installed():
+    from pyspark.sql import SQLContext, SparkSession
+    from pyspark.ml.feature import SQLTransformer
+
+    spark = SparkSession.builder.getOrCreate()
+    sql = SQLContext(spark.sparkContext)
+
+if pandas_installed():
+    import pandas as pd
+
+
+class TestSparkMLSQLTransformer(unittest.TestCase):
+    # Test SQLTransformer
+    @unittest.skipIf((not sparkml_installed()) or (not pandas_installed()), reason="Spark-ML test requires pyspark and pandas")
+    @unittest.skipIf(LooseVersion(torch.__version__) < LooseVersion("1.6.0"), reason="Spark-ML test requires torch >= 1.6.0")
+    def test_sql_transformer_converter(self):
+        iris = load_iris()
+        features = ['sepal_length', 'sepal_width', 'petal_length', 'petal_width']
+
+        pd_df = pd.DataFrame(data=np.c_[iris['data'], iris['target']], columns=features + ['target'])[["sepal_length", "sepal_width", "petal_length", "petal_width"]]
+        df = sql.createDataFrame(pd_df)
+
+        model = SQLTransformer(statement="SELECT *, sepal_length*(sepal_length/sepal_width) as new_feature1,"
+                                         " petal_length*(petal_length/petal_width) as new_feature2,"
+                                         " petal_length + 10.0*(sepal_width - petal_width) as new_feature3 from __THIS__")
+
+        output_col_names = ['new_feature1', 'new_feature2', 'new_feature3']
+        test_df = df
+        torch_model = convert(model, "torch", test_df)
```

Comment:
One thing that I noticed in other Spark tests is that we use `test_df` at conversion time (i.e., a Spark `DataFrame`) and a regular pandas `DataFrame` at inference. We should make sure that also inference works over Spark DataFrames (not necessarily in this PR).

Reply:
Agree! However, I think that will require an integration of HB with Spark to distribute the inference using Spark on Spark DataFrames. Also, I think we should support the Pandas data frame at conversion time. Some people may only have the serialized Spark ML pipeline and not the original data frame.

Reply:
Yes I agree. Actually that would be simpler, right? We can just assume that the input of the converter \ transformation is a Pandas DataFrame, and call it done. Perhaps we can add an exception in the case it's a Spark DataFrame asking to convert it a Pandas DataFrame.